bluesky/run_engine.py

```diff
@@ -270,7 +270,7 @@ def _clear_run_cache(self):
         self._metadata_per_run.clear()
         self._bundling = False
         self._run_is_open = False
-        self._msg_cache = None  # checkpoints can't rewind into a closed run
```

Comment:
This could be `self._run_start_uid = None` if you prefer. The point is to erase the old uid -- might as well make the next one while we are here. But now that I say this I think `None` is safer.